File: backend/mobile-leetnote-backend/src/main/java/com/example/leetnote_backend/model/DTO/EvaluationDetailDTO.java
package com.example.leetnote_backend.model.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class EvaluationDetailDTO {
    private Long evaluationId;
    private Long problemId;
    private String problemTitle;
    private String difficulty;
    private LocalDateTime createdAt;
    private EvaluationDTO evaluation;
    private String solutionText;
}

File: frontend/mobile-leetnode-frontend/app/src/main/java/com/example/leetnote/data/model/EvaluationDetailDTO.kt
package com.example.leetnote.data.model

data class EvaluationDetail(
    val id: Long,
    val version: Int,
    val evaluation: EvaluationDTO,
    val createdAt: String
)

data class EvaluationDTO(
    val rating: Int,
    val issue: List<String>,
    val feedback: List<String>
)

data class EvaluationDetailDTO(
    val evaluationId: Long,
    val problemId: Long,
    val problemTitle: String,
    val difficulty: String,
    val createdAt: String,
    val evaluation: EvaluationDTO,
    val solutionText: String
)

data class EvaluationListItemDTO(
    val evaluationId: Long,
    val problemId: Long,
    val problemTitle: String,
    val createdAt: String,
)